### PR TITLE
Usability improvements

### PR DESCRIPTION
**Pull request by jtoberling**
- Improvement mode toml filelist have to be in full-path sorted mode. 
  By using large project the current listing is "mixed" and very hard to use. 

- Increasing timeout value in chat_to_files finditer. 
  By using DeepSeek API with "deepseek-coder" model without this timeout increment the resolution of completion is timing out quite frequently.

- Adding "--skip-file-selection" / "-s" command line option to skip toml file selection editor in improvement mode.
  By using it in complex projects and repetitive tasks the file selector popup is especially disturbing and against usabilty. 

Please consider to merge the above points as they are leading to a more smooth usability experiment.

### CODE REVIEW

**Review by similato87 — CHANGES_REQUESTED** [review]
Great job overall! The implementation looks solid, and the functionality is well thought out. The main thing needed now is to add tests. You can find the test cases [test_main.py](https://github.com/gpt-engineer-org/gpt-engineer/blob/main/tests/applications/cli/test_main.py). Since there isn't a specific test case for the file selector, please write some tests for the main process to ensure everything passes.

**Comment by similato87** [comment]
> * Improvement mode toml filelist have to be in full-path sorted mode.
>   By using large project the current listing is "mixed" and very hard to use.
> * Increasing timeout value in chat_to_files finditer.
>   By using DeepSeek API with "deepseek-coder" model without this timeout increment the resolution of completion is timing out quite frequently.
> * Adding "--skip-file-selection" / "-s" command line option to skip toml file selection editor in improvement mode.
>   By using it in complex projects and repetitive tasks the file selector popup is especially disturbing and against usabilty.
> 
> Please consider to merge the above points as they are leading to a more smooth usability experiment.

@jtoberling Thank you for your meaningful contributions! Your efforts have significantly impacted us, and we truly appreciate your dedication. We love having talented and passionate contributors like you on the team. Keep up the great work!

File: gpt_engineer/core/chat_to_files.py
[the remark this answers is not in the document]
Comment (jtoberling) [reply gpt_engineer/core/chat_to_files.py:1]: Defaulted to 3 secs. 
Added a new command line parameter "--diff_timeout INTEGER" to override. 
Also created some test in tests/application/cli/test_main.py:test_improve_existing_project_diff_timeout for the new command line option.

File: gpt_engineer/applications/cli/main.py
[the remark this answers is not in the document]
Comment (jtoberling) [reply gpt_engineer/applications/cli/main.py:1]: Added test:
-  tests/application/cli/test_main.py:test_improve_existing_project_skip_file_selection

File: gpt_engineer/applications/cli/file_selector.py
[the remark this answers is not in the document]
Comment (jtoberling) [reply gpt_engineer/applications/cli/file_selector.py:1]: Added tests:
- tests/core/test_file_selector_enhancements.py: test_file_selector_enhancement_skip_file_selector
- tests/core/test_file_selector_enhancements.py: test_file_selector_enhancement_sort

File: gpt_engineer/tools/supported_languages.py
[the remark this answers is not in the document]
Comment (jtoberling) [reply gpt_engineer/tools/supported_languages.py:1]: Also added Arduino *.ino files. They are basically C++ with some extra functions like "setup()" and  "loop()" - works well with ChatGPT & DeepSeek.

File: gpt_engineer/core/chat_to_files.py
[the remark this answers is not in the document]
Comment (similato87) [reply gpt_engineer/core/chat_to_files.py:1]: Great work on defaulting time and adding the new command line parameter for overriding. The test you created for this is a nice addition.

File: gpt_engineer/applications/cli/file_selector.py
[the remark this answers is not in the document]
Comment (similato87) [reply gpt_engineer/applications/cli/file_selector.py:1]: Thanks for adding the tests for the file selector enhancements. I appreciate your thoroughness in testing both skipping and sorting.

File: gpt_engineer/applications/cli/main.py
[the remark this answers is not in the document]
Comment (similato87) [reply gpt_engineer/applications/cli/main.py:1]: Valuable enhancement. Thanks for covering this aspect.

File: gpt_engineer/tools/supported_languages.py
[the remark this answers is not in the document]
Comment (similato87) [reply gpt_engineer/tools/supported_languages.py:1]: Including the Arduino *.ino files was a smart move. It's good to know they integrate well with ChatGPT & DeepSeek.

**Comment by similato87** [comment]
@jtoberling I fully agree with the changes you’ve made and approve this PR. We'll discuss this further in our next group meeting. Great job!

**Review by similato87 — APPROVED** [review]
I fully agree with the meaningful enhancements for skipping file selection and adding buffer time for large codebases. The extended support for more languages, including Arduino files, is a great addition. Everything is well-designed and clearly tested.

**Pull request merged**